editor/sidebar/block-inspector/advanced-controls.js

```diff
@@ -69,7 +69,12 @@ class BlockInspectorAdvancedControls extends Component {
 		}
 
 		return (
-			<PanelBody className="editor-advanced-controls" title={ __( 'Advanced' ) }>
+			<PanelBody
+				className="editor-advanced-controls"
+				initialOpen={ false }
+				key={ `adavanced-controls-${ selectedBlock.uid }` }
```

Comment:
Shouldn't be `key` applied in the place where this component is rendered? Which mean [here](https://github.com/WordPress/gutenberg/blob/b53887488b2de0ff28a9aa9c6e05f5b4c3c95af2/editor/sidebar/block-inspector/index.js#L33). Keys are usually applied when rendering sibling components, rather than inside the child component. Just checking as I'm not quite sure why this key is necessary here.

Reply:
Yes you are totally right in this case it is better to add the key there. Keys are necessary because without keys react reuses the component and keeps the state, so initialOpen become irrelevant react uses that state that comes from the component before.

Slot fill then assigns new keys 0..1..2 to the components passed to it but not to its children so sometimes there is a new to set keys inside child components. In this case, I checked that slot fill is not being used so there is no problem in passing the key in the correct place.